Networking/Networking/Remote/SiteVisitStatsRemote.swift

```diff
@@ -48,3 +53,37 @@ private extension SiteVisitStatsRemote {
         static let statFields: String = "stat_fields"
     }
 }
+
+/// An error that occurs at site visit stats networking layer.
+/// API documentation of possible errors:
+/// https://developer.wordpress.com/docs/api/1.1/get/sites/%24site/stats/
+///
+public enum SiteVisitStatsRemoteError: Error {
+    case statsModuleDisabled
+    case unknown
+
+    private enum ErrorIdentifiers {
+        static let invalidBlog: String = "invalid_blog"
+    }
+
+    private enum ErrorMessages {
+        static let statsModuleDisabled: String = "This blog does not have the Stats module enabled"
+    }
+
+    init(error: Error) {
+        guard let dotcomError = error as? DotcomError else {
+            self = .unknown
+            return
+        }
+        switch dotcomError {
+        case .unknown(let code, let message):
+            if code == ErrorIdentifiers.invalidBlog && message == ErrorMessages.statsModuleDisabled {
```

Comment:
I would probably pass up the error as it is, without trying to figure out if it is due to the plugin being disabled, and do this very same check in Yosemite. On the one hand, it seems to me like deciding the type of error that we receive, and what that error means to our system, is by definition business logic, and the networking layer should only move data to and from the backend, but on the other hand, maybe I'm overthinking it, what do you think?

Reply:
We already have a [little bit of business logic](https://github.com/woocommerce/woocommerce-ios/blob/develop/Networking/Networking/Remote/Remote.swift#L71-L75) in the main `Remote` class, so I would say there is precedent for this. You are our architect, so if you have your own preferences, let's follow that :)

Reply:
Fair enough!

Reply:
My reason for classifying the error at the Networking layer was that the error definition is from the API and translating the API error into what the API doc specifies felt like a networking concern. On the other hand, I see that this is business logic and other remote services don't seem to handle the errors at Networking layer. I'll move this to Yosemite layer!

Reply:
Moved the error handling in this commit: https://github.com/woocommerce/woocommerce-ios/pull/1105/commits/a62110d618728519536ca6ce396df2995cce92f5

Reply:
Thanks for taking the time to take the feedback into consideration!